_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,865 @@
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: jhipster-role
+subjects:
+  - namespace: demo
+    kind: ServiceAccount
+    name: jhipster-service-account
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: jhipster-service-account
+  namespace: demo
+```
+
+Notice there are some differences with the Spinnaker service account. The k8s manifests generated by JHipster include the creation of `secrets`, `configmaps`, and `persistentvolumeclaims`, and the management of those Kubernetes resources must be granted to the service account used for deploying the JHipster k8s manifests.
+
+Create the Kubernetes `jhipster-service-account`:
+
+```shell
+CONTEXT=$(kubectl config current-context)
+
+kubectl apply --context $CONTEXT \
+    -f ./jhipster-service-account.yml
+
+
+TOKEN=$(kubectl get secret --context $CONTEXT \
+   $(kubectl get serviceaccount jhipster-service-account \
+       --context $CONTEXT \
+       -n demo \
+       -o jsonpath='{.secrets[0].name}') \
+   -n demo \
+   -o jsonpath='{.data.token}' | base64 --decode)
+
+kubectl config set-credentials ${CONTEXT}-token-user --token $TOKEN
+
+kubectl config set-context $CONTEXT --user ${CONTEXT}-token-user
+```
+
+Add the account to the Halyard configuration:
+
+```shell
+CONTEXT=$(kubectl config current-context)
+
+hal config provider kubernetes account add jhipster-gke-account \
+    --context $CONTEXT
+```
+
+### Add a docker registry account
+
+As the goal is to trigger the pipeline execution when a new image is pushed to DockerHub, you need to configure a docker-registry provider with Halyard.
+
+First, enable the docker-registry provider:
+
+```shell
+hal config provider docker-registry enable
+```
+
+The following `hal config` line will prompt for your password or access token if you have 2FA enabled. Make sure to replace `your-dockerhub-username`.
+
+```shell
+ADDRESS=index.docker.io
+USERNAME=your-dockerhub-username
+
+hal config provider docker-registry account add docker-account \
+    --address $ADDRESS \
+    --repositories $USERNAME/store $USERNAME/gateway \
+    --username $USERNAME \
+    --password
+```
+
+### Add a GitHub artifact account
+
+The pipeline will deploy k8s manifests from a GitHub repository, so you must also configure a GitHub artifact account. First, enable the feature:
+
+```shell
+hal config artifact github enable
+```
+Generate an access token on GitHub. Sign in, and in the user menu choose **Settings**. Then on the left menu choose **Developer settings**. On the left menu choose **Personal access tokens**. Click **Generate new token** and copy the token.
+
+Then add your GitHub username and set the new token when prompted:
+
+```shell
+ARTIFACT_ACCOUNT_NAME=your-github-username
+
+hal config artifact github account add $ARTIFACT_ACCOUNT_NAME \
+    --token
+```
+
+The new GKE, Docker, and GitHub accounts configuration must be applied to the deployment before starting the pipeline design:
+
+```shell
+hal deploy apply
```

Comment:
This is a little scary:

```
Validation in default.security:
- WARNING Your UI or API domain does not have override base URLs
  set even though your Spinnaker deployment is a Distributed deployment on a
  remote cloud provider. As a result, you will need to open SSH tunnels against
  that deployment to access Spinnaker.
? We recommend that you instead configure an authentication
  mechanism (OAuth2, SAML2, or x509) to make it easier to access Spinnaker
  securely, and then register the intended Domain and IP addresses that your
  publicly facing services will be using.
```

Reply:
Should we configure the authentication for Spinnaker in this post or just add a note about security?

Reply:
I think this post is already long enough. I'd just add a note and maybe link to their docs. https://spinnaker.io/docs/setup/other_config/security/authentication/oauth/

It doesn't seem like the Spinnaker instance is public, is it?